### ISSUE TITLE
Bus in Inference Code with beam search

### ISSUE THREAD

**Issue by Lyman-Smoker** [szqwu/Motion-Agent#10]
It seems that there are some bugs in the `generate(self, caption)` function in https://github.com/szqwu/Motion-Agent/blob/efc191de51ae049d975e63cb36b63c38a1bdf557/models/mllm.py#L119C9-L134C86

Specifically, the motion generation process goes like:

        outputs = self.llm.generate(
            input_ids, 
            max_length=200, 
            num_beams=2, 
            early_stopping=True, 
            return_dict_in_generate=True, 
            output_scores=True
        )

        scores = torch.stack(outputs.scores)  # [num_generated_tokens, num_beams, vocab_size]
        # print(scores.shape)
        # Take only the best beam (beam 0)
        best_beam_scores = scores[:, 0, :]  # [num_generated_tokens, vocab_size]
        motion_logits = best_beam_scores[:, -(self.args.nb_code+2):]
        # print(motion_logits.shape)
        motion_tokens = torch.argmax(motion_logits, dim=-1)  # [num_generated_tokens]

However, when using beam search, it seems that we are not able to get the best beam sequence by simply calling `torch.stack(outputs.scores)[:, 0, :]`, which means the extracted motion from this function is not consistent with the motion sequence from `outputs.seuqences`. To address this, using `outputs.beam_indices` seems necessary.

Besides, I have tried to simply use greedy search during inference, and I notice that the model suffers from obvious performance drop.

**Comment by CCCloth** [comment]
@Lyman-Smoker I'm having the same issue. I tried using `outputs.beam_indices` to recover the motion tokens, but the results became even worse. Additionally, could you please confirm whether you were able to reproduce the text-to-motion performance reported in the paper on the HumanML3D test set? I used the provided models (`motionllm.pth` and `vqvae.pth`) along with the evaluation code from the authors, but my results are still far from those reported in the paper.

**Comment by Lyman-Smoker** [comment]
> [@Lyman-Smoker](https://github.com/Lyman-Smoker) Thank you very much, this is really helpful for me! By the way, I'd like to ask, how many epochs did you train your model on the `t2m` task to achieve this result? Or do you still remember what the `loss` and `accuracy` were? I truly appreciate your time and support!

@CCCloth Sorry for the late reply.

Note that the previously provided results are from the official checkpoint. 

However, I am also not able to reproduce the posted results in paper by training my own MotionLLM from scratch. 
In my experiments, training MotionLLM from scratch takes around 70 epochs to converge, and the accuracy at 70 epoch is around 80.0. Continuing the training brings higher accuracy, but does not bring clearly higher generation results.

I have also trained a MotionLLM from scratch on the KIT-ML dataset, the results are shown as follows:
```
Top-1: 0.416 \pm .004
Top-3: 0.746 \pm .003
FID: 0.560 \pm .025
MM-Dist: 3.001 \pm .011
Diversity: 11.209 \pm .135 
```

On the post training, in my experiments, training MotionLLM (initialized with the official checkpoint) on T2M task does not bring clear performance improvement. Specifically, the Top-k and MM-Dist score drop but FID score improves. I hypothesize it is because the model is already converged on the T2M task in previous training and tends to over-fit the training data when further trained on the same task instead of learning new text-motion alignment knowledge. 

However, I have to note that MotionLLM has potential for further improvement by learning new tasks requiring stronger text-motion alignment.

Hopefully the authors could provide some explanations on the previously mentioned bugs and the reproduced results. 
Moreover, If there is anything wrong on my side, or you have opposite results and conclusions with mine, please tell me. Thank you so much :)
